src/matlab2tikz.m

```diff
@@ -3543,60 +3543,144 @@ function warnFacetedInterp(m2t)
 end
 % ==============================================================================
 function [m2t, str] = drawScatterPlot(m2t, h)
+    % DRAWSCATTERPLOT Draws a scatter plot
+    %
+    % A scatter plot is a plot containing only markers and where the
+    % size and/or color of each marker can be changed independently.
+    %
+    % References for TikZ code:
+    %  - http://tex.stackexchange.com/questions/197270/how-to-plot-scatter-points-using-pgfplots-with-color-defined-from-table-rgb-valu
+    %  - http://tex.stackexchange.com/questions/98646/multiple-different-meta-for-marker-color-and-marker-size
+    %
+    % See also: scatter
     str = '';
     if ~isVisible(h)
         return; % there is nothing to plot
     end
-
-    xData = get(h, 'XData');
-    yData = get(h, 'YData');
-    zData = get(h, 'ZData');
-    cData = get(h, 'CData');
-    sData = get(h, 'SizeData');
-
-    if isempty(cData) && strcmpi(getEnvironment(), 'Octave')
-        cData = get(h, 'MarkerEdgeColor');
+    
+    dataInfo   = getDataInfo(h, 'X','Y','Z','C','Size');
+    markerInfo = getMarkerInfo(m2t, h);
+    
+    if isempty(dataInfo.C) && strcmpi(getEnvironment(), 'Octave')
+        dataInfo.C = get(h, 'MarkerEdgeColor');
     end
 
-    markerInfo = getMarkerInfo(m2t, h);
     %TODO: check against getMarkerOptions() for duplicated code
```

Comment:
It is still applicable: both functions are nontrivial and way too complex to see immediately whether they do the same or not. I.e. when refactoring those, we should check for duplication and maybe even try to merge them.